automation/entry_scripts/balena-deploy-block.sh

```diff
@@ -23,28 +25,34 @@ if [ "$ESR" = "true" ]; then
 	APPNAME="${APPNAME}-esr"
 fi
 
-echo "[INFO] Pushing $_local_image to ${BALENAOS_ACCOUNT}/$APPNAME"
+echo "[INFO] Deploying  to ${BALENAOS_ACCOUNT}/$APPNAME"
 balena_api_create_public_app "${APPNAME}" "${BALENARC_BALENA_URL}" "${MACHINE}" "${balenaCloudEmail}" "${balenaCloudPassword}" "${BOOTABLE}"
-_releaseID=$(balena deploy "${BALENAOS_ACCOUNT}/$APPNAME" "$_local_image" | sed -n 's/.*Release: //p')
+_releaseID=$(balena_lib_release "${BALENAOS_ACCOUNT}/$APPNAME" "${FINAL}" "/deploy" "${API_ENV}" "$_local_image")
+if [ -z "${_releaseID}" ]; then
+	echo "[INFO] Failed to deploy to ${BALENAOS_ACCOUNT}/$APPNAME"
+	exit 1
+fi
 
 # Legacy hostapp tagging
-release_version="${RELEASE_VERSION}"
-if [ -n "${VARIANT}" ]; then
+if [ "${FINAL}" = "yes" ] && [ -n "${VARIANT}" ]; then
```

Comment:
So all builds will get deployed as draft releases (even PRs ?) unless the build has the `deploy` jenkins build param set to `yes` in which case the release is deployed as final?
And the deployment as final should only happen at this time when the `BOARDNAME` jenkins job has been called by `yocto-deploy-board` job.

Reply:
@floion in current releases a hostapp is only deployed when the job is marked as `deploy`.

With the current state of this PR, this does not change. Hostapps are only deployed if the job is marked as `deploy. And as you well say, all deployments will be draft releases. They will need to be flagged as final externally, either manually or by Leviathan.

At the moment this has little implications as balenaCloud does not yet use the release version but the hostapp labels, and these are created for `deploy` jobs.